src/nupic/regions/SPRegion.cpp

```diff
@@ -121,13 +120,13 @@ void SPRegion::initialize() {
   if (args_.potentialRadius == 0)
     args_.potentialRadius = args_.inputWidth;
 
-  sp_ = new SpatialPooler(
+  sp_ = std::unique_ptr<SpatialPooler>(new SpatialPooler(
```

Comment:
👍 
Keep in mind that this makes SPRegion non-copyable.  I think that is ok.

Reply:
The smart pointer leads to some more errors, 

> [          ] Setup Network; add 3 regions and 2 links.
ERR:  CHECK FAILED: "srcOutput->getDataType() == destInput->getDataType()" Network::link -- Mismatched data types.Real32 != UInt32 [/home/travis/build/htm-community/nupic.cpp/src/nupic/engine/Network.cpp line 312]
unknown file: Failure
C++ exception with description "CHECK FAILED: "srcOutput->getDataType() == destInput->getDataType()" Network::link -- Mismatched data types.Real32 != UInt32" thrown in the test body.
[  FAILED  ] SPRegionTest.testLinking (0 ms 

W/o it the tests cleanly pass.

Reply:
Oh, the network in master does not have my changes so types must still match.

Reply:
> network in master does not have my changes so types must still match.

Can you point me where the datatype of the field (srcOutput, destInput) is defined?